### PR TITLE
Export Fix

### PR DESCRIPTION
**Pull request by rodriguez-jay**
Added mpi barriers to the ends of mesh and physics exports to resolve issue with visualizing extruded geometries in visit and paraview. Issue #132.

### CODE REVIEW

File: lua/framework/physics/field_function/physics_export_field_func.cc [hunk 1]
```diff
@@ -29,6 +30,7 @@ ExportFieldFunctionToVTK(lua_State* L)
   //  ff->ExportToVTK(base_name);
   FieldFunctionGridBased::ExportMultipleToVTK(base_name, {ff});
 
+  opensn::mpi_comm.barrier();
```
Comment (andrsd) [inline lua/framework/physics/field_function/physics_export_field_func.cc:33]: The same business here. Move the barrier inside ` FieldFunctionGridBased::ExportMultipleToVTK`. This way you fix all the usage of that method.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by rodriguez-jay** [comment]
Changes added and pushed into my fork.

**Comment by andrsd** [comment]
Now we need to fix your "branch" commit history. As you see you are trying to merge 20 commits while you only did 2. The  mistake you made was that you started to develop on `main`. Do not do that. Your local `main` should follow upstream `main` (i.e. main in `Open-Sn/opensn` repo). When working on code, for your own sanity, create a branch and do your development on it (we call such a branch a "feature branch"). This will allow you to manipulate your work (we call that rebasing) when you receive a review or some other work makes it into the `main` that would create conflicts with your branch. 

This is would you do now to fix your PR:
- `git checkout main`
   This is just a sanity step, so we are sure the next steps will do the right thing.
- `git pull --rebase upstream main` 
   Here we assume that you have a remote called `upstream` that points to Open-Sn/opensn. IDK how you cloned your local fork, so this is the only advice I can give you. For this command to work you may need to do this: `git remote add upstream git@github.com:Open-Sn/opensn.git`
- `git checkout -b issue/132` 
   This is creating a feature branch called `issue/132`. This branch should be 2 commits ahead of upstream `main`.
- `git rebase --interactive HEAD~2` 
   This will open an editor and you will see something like this:
   ```
   pick <sha1hash> Export Fix
   pick <sha1hash> Export Fix Changes
   ```
   Change the second pick to `f`, so it looks like this:
   ```
   pick <sha1hash> Export Fix
   f <sha1hash> Export Fix Changes
   ```
   Save the file and exit the editor. This will merge the second patch into the first one.

- Now, push the branch into your fork and open a PR into our `main`. Use the same procedure you used for opening this PR. Double check that there is only a single commit in your PR (github will show that to you when opening a PR). This is a check for you that you did the above steps correctly.

- Then, close this PR, since we will use the new one.

- Fix your local `main`.
   - `git checkout main`
   - `git reset --hard HEAD~2`

Last, but not least. Do not merge `main`  into your feature branches...

**Comment by rodriguez-jay** [comment]
Thanks for the help David! I really appreciate the instructions. I made the changes. If there is anything else for me to do, please let me know.

**Comment by andrsd** [comment]
No problem! Closing this in favor of #142

**Pull request closed without merging**